src/core/tests/Test_PatchInstaller.py

```diff
@@ -177,16 +178,24 @@ def healthstore_writes_helper(self, health_store_id, maintenance_run_id, expecte
         argument_composer.maintenance_run_id = maintenance_run_id
         runtime = RuntimeCompositor(argument_composer.get_composed_arguments(), True, Constants.YUM)
 
+        if is_force_reboot:
+            runtime.package_manager.force_reboot = True
+
         runtime.set_legacy_test_type('SuccessInstallPath')
         installed_update_count, update_run_successful, maintenance_window_exceeded = runtime.patch_installer.install_updates(
             runtime.maintenance_window, runtime.package_manager, simulate=True)
         runtime.patch_installer.mark_installation_completed()
 
         with runtime.env_layer.file_system.open(runtime.execution_config.status_file_path, 'r') as file_handle:
-            substatus_file_data = json.load(file_handle)[0]["status"]["substatus"][1]
+            substatus_file_data = json.load(file_handle)[0]["status"]["substatus"]
+
+        if is_force_reboot:
+            self.assertEqual('warning', substatus_file_data[0]['status'])
+        else:
+            self.assertEqual('success', substatus_file_data[0]['status'])
 
-        self.assertEqual(True, json.loads(substatus_file_data['formattedMessage']['message'])['shouldReportToHealthStore'])
-        self.assertEqual(expected_patch_version, json.loads(substatus_file_data['formattedMessage']['message'])['patchVersion'])
+        self.assertEqual(True, json.loads(substatus_file_data[1]['formattedMessage']['message'])['shouldReportToHealthStore'])
```

Comment:
Why did the index change here? What was added at 0th index?

Reply:
this is helper func that used for 5 healthstore utests, and i want to reuse the set up. the substatus is a list of object and at index 0 it has the status field which needs to be test either warning or success or other state. if substatus_file_data is at index 1, then i will not able to get the status value

![image](https://user-images.githubusercontent.com/127874208/236576604-c95b65bf-284d-444f-8913-096200241d9e.png)

Reply:
Did not quite understand this. From what I get here is: substatuses are PatchAssessmentSummary, PatchInstallationSummary, PatchMetadatForHealthStore and ConfigurePatchingSummary. There aren't any changes to these overall substatuses', just "status" field within PatchInstallationSummary. So if these substatuses already exists in the result, why would their index change with a change explicitly to the "status" field within "PatchInstallationSummary" substatus?

Reply:
Adding to the comment above, the position (index) of these substatuses within substatus_file_data should not change because the helper function is not changing that

Reply:
sorry that image was a bad example, here's the actual substatus_file result from the test:
![image](https://user-images.githubusercontent.com/127874208/236581287-4748e5ee-12b0-46a0-9aa8-ed7faffc833e.png)
as you can at index 0 is patch installation, if I keep it at index 1 then i am looking at the patchmetahealth section

Reply:
The question is why did PatchMetadataFromHealthStore go from index 0 to 1? 

Your change in healthstore_writes_helper() set runtime.package_manager.force_reboot to True based on the incoming is_force_reboot function parameter. After which, in line 198-190, it is reading the contents of from status_file_path.
So there is no explicit file write in this code change (the write part happens inherently in the background, like before).

So to answer the question on why did PatchMetadataFromHealthStore go from index 0 to 1, we need to find these things:
1. Did the previous test (i.e. before this change) not have PatchInstallationSummary substatus at all? 
2. Or it did and was at a different index? In that case, why did the indices change? (ideally should not have )
3. If PatchInstallationSummary wasn't there before, why so and why is it added now?

Reply:
comments and questions are resolved in our codebase meeting, it's okay to set index at 1 in assert()